Home task completed

DChristina opened the pull request.

DChristina committed:
Home task completed

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,64 @@
+    private Student groupHead ;
+    private List<Student> groupMembers = new ArrayList<>();
+    private List<String> tasksForGroup = new ArrayList<>();
+
```
Add structure `Map<String, List<Student>> compleatedTasks = new HashMap<>();` to mark compleaded tasks for students

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public void changeGroupHead(Student newGroupHead){
+        this.groupHead = newGroupHead;
```
You do not check that new head is in students list

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,64 @@
+        tasksForGroup.add(task);
+    }
+
+
```
add method to mark task for student as compleated

DChristina replied:
I already had such a method  in Student class
public void markTaskAsMadeByStudent (String task,Boolean isDone){
       statesOfTasks.put(task,isDone);
    }
But I add additional method for all group.

DChristina committed:
Fixed bugs according comments in merge request.
New markTaskAsCompletedByStudentOfTheGroup method was added, updated several methods (addTaskForGroup,deleteGroupMember,addGroupMember,changeGroupHead).

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    public List<Student> getGroupMembers() {
+        return groupMembers;
```
Copy lists before returning

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    public List<String> getTasksForGroup() {
+        return tasksForGroup;
```
Copy lists before returning

yurii-voronenko commented on src/Group.java:
```diff
@@ -0,0 +1,91 @@
+            } else {
+                for (Student searchedStudent :groupMembers){
+                    if(deletedStudent==searchedStudent){
+                        groupMembers.remove(deletedStudent);
```
You can not do this in foreach statement you will get exception

yurii-voronenko commented on src/Student.java:
```diff
@@ -0,0 +1,49 @@
+
+     void markTaskAsMadeByStudent (String task,Boolean isDone){
+
+        statesOfTasks.put(task,isDone);
```
Check task on null before add it to map

yurii-voronenko reviewed (approved):
LGTM
